src/libxtp/gwbse/bse.cc

```diff
@@ -51,7 +53,30 @@ void BSE::SetupDirectInteractionOperator() {
         << ctp::TimeStamp() << " Setup TDA triplet hamiltonian " << flush;
       CTP_LOG(ctp::logDEBUG, _log)
         << ctp::TimeStamp() << " Solving for first "<<_opt.nmax<<" eigenvectors"<< flush;
-      tools::linalg_eigenvalues(H , _bse_triplet_energies, _bse_triplet_coefficients ,_opt.nmax );
+
+      if (_opt.davidson)
+      {
+        CTP_LOG(ctp::logDEBUG, _log)
+        << ctp::TimeStamp() << " Using Davidson Iterative Eigen Solver " << flush;
+        DavidsonSolver DS(_log);
+        if (_opt.jocc)
+        {
+          DS.set_jacobi_correction();
+          DS.set_jacobi_linsolve(_opt.jocc_linsolve);
+        }
+        DS.solve(H,_opt.nmax);
+        _bse_triplet_energies = DS.eigenvalues().cast<float>();
```

Comment:
this will break if compiled with double. Just stay in double at the moment, e.g. compile your code with cmake -DGWBSE_DOUBLE=1 ..